src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -175,13 +175,15 @@ internal void CloseAsIs()
                     SpinWait sw = new SpinWait();
                     while (!_released)
                     {
+                        // The socket was not released due to the SafeHandle being used.
+                        // Try to make those on-going calls return.
+                        innerSocket.TryUnblockSocket(abortive);
```

Comment:
Calling this more than once will help?

Reply:
I think you could end up with something like:

thread1: increments SafeHandle
thread2: Dispose - TryUnblockSocket
thread1: operation is using handle